app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> None:
+    result_dict = {}
+
+    def wrapper(*args) -> Any:
+        if args in result_dict:
+            print("Getting from cache")
+            return result_dict[args]
+        else:
+            result_dict.update({args: func(*args)})
+            print("Calculating new result")
+            return result_dict[args]
```

Comment:
These retunrs are breaking the DRY principle. I bet you can somehow avoid such code duplication...